### PR TITLE
Added sanity checks, fixed url join

### PR DESCRIPTION
**Pull request by skhomuti**

### CODE REVIEW

File: src/providers/consensus/client.py [hunk 1]
```diff
@@ -35,6 +36,16 @@ class ConsensusClient(HTTPProvider):
     API_GET_BLOCK_HEADER = '/eth/v1/beacon/headers/{}'
```
Comment (avsetsin) [inline src/providers/consensus/client.py:36]: I would remove `/` at the start to be the same as the rest of endpoints

Comment (skhomuti) [reply src/providers/consensus/client.py:36]: nice catch!

File: src/modules/submodules/consensus.py [hunk 2]
```diff
@@ -40,6 +40,20 @@ def __init__(self, w3: Web3):
...
             raise NotImplementedError('CONTRACT_VERSION and CONSENSUS_VERSION should be set.')
 
+    def check_sanity(self):
+        root = self.w3.cc.get_block_root('head').root
+        block_details = self.w3.cc.get_block_details(root)
+        bs = build_blockstamp(block_details)
+
+        config = self.get_chain_config(bs)
+        cc_config = self.w3.cc.get_config_spec()
+        if any((config.genesis_time != cc_config.MIN_GENESIS_TIME,
```
[the remark this answers is not in the document]
Comment (skhomuti) [reply src/modules/submodules/consensus.py]: Fixed, thanks

**Review by avsetsin — APPROVED** [review]

File: src/providers/consensus/typings.py [hunk 3]
```diff
@@ -6,6 +6,28 @@
 from src.utils.dataclass import Nested, FromResponse
 
 
+@dataclass
+class BeaconSpecResponse(FromResponse):
+    DEPOSIT_CHAIN_ID: int
+    SLOTS_PER_EPOCH: int
+    SECONDS_PER_SLOT: int
+
+    def __post_init__(self):
```
Comment (F4ever) [inline src/providers/consensus/typings.py:15]: It can not be used as frozen. Is there any way to do it in __new__ and maybe do this in from_reponse?

Comment (skhomuti) [reply src/providers/consensus/typings.py:15]: Not sure whether we should solve it in the current PR

File: src/providers/http_provider.py [hunk 4]
```diff
@@ -60,8 +66,13 @@ def _get(self, url: str, params: Optional[dict] = None) -> Tuple[dict | list, di
 
         try:
             json_response = response.json()
-            data = json_response['data']
-            del json_response['data']
+            if 'data' in json_response:
+                data = json_response['data']
```
Comment (F4ever) [inline src/providers/http_provider.py:70]: Move to else

File: src/providers/http_provider.py [hunk 5]
```diff
@@ -60,8 +66,13 @@ def _get(self, url: str, params: Optional[dict] = None) -> Tuple[dict | list, di
...
+                del json_response['data']
+                meta = json_response
+            else:
+                data = json_response
+                meta = {}
         except (KeyError, JSONDecodeError) as error:
```
[the remark this answers is not in the document]
Comment (skhomuti) [reply src/providers/http_provider.py]: yes sir

**Review by F4ever — APPROVED** [review]

**Pull request merged**